File: seller/api/model/shipment.go
package model

import (
	"time"
)

type Shipment struct {
	TrackingCode  string    `json:"trackingCode"`
	Courier       string    `json:"courier"`
	Sender        Sender    `json:"sender"`
	Recipient     Recipient `json:"recipient"`
	WeightInGrams int       `json:"weightInGrams"`
	Size
	Content             string     `json:"content"`
	ShippingType        string     `json:"shippingType"`
	IsFragile           bool       `json:"isFragile"`
	LastState           string     `json:"lastState"`
	LastLocation        *Address   `json:"lastLocation"`
	IsInCourierFacility bool       `json:"isInCourierFacility"`
	CreatedAt           *time.Time `json:"createdAt"`
	UpdatedAt           *time.Time `json:"updatedAt"`
}

type ShipmentRecord struct {
	Key   string `json:"key"`
	Value string `json:"value"`
}

type Shipments []ShipmentRecord

File: seller/web/src/app/core/models/shipment.ts
export interface Shipment {
  trackingCode: string;
  courier: string;
  sender: {
    firstName: string;
    lastName: string;
    country: string;
    city: string;
    postCode: string;
    address: string;
    phone: string;
  },
  recipient: {
    firstName: string;
    lastName: string;
    country: string;
    city: string;
    postCode: string;
    address: string;
    phone: string;
  },
  weightInGrams: number;
  heightInMM: number;
  widthInMM: number;
  depthInMM: number;
  content: string;
  shippingType: string;
  isFragile: boolean;
  lastState: string;
  lastLocation?: {
    country: string;
    city: string;
    postCode: string;
    address: string;
  };
  isDelivered: boolean;
  createdAt: Date;
  updatedAt: Date;
}
